src/devices/src/legacy/rtc_pl031.rs

```diff
@@ -1,336 +1,73 @@
-                RTCCR => 1, // RTC is always enabled.
-                RTCIMSC => self.imsc,
-                RTCRIS => self.ris,
-                RTCMIS => self.ris & self.imsc,
-                off => {
-                    return Err(Error::BadOffset(off, "read"));
-                }
-            };
-        }
-        Ok(val)
-    }
-
-    fn handle_write(&mut self, offset: u64, val: u32) -> Result<()> {
-        match offset {
-            RTCMR => {
-                // The MR register is used for implementing the RTC alarm. A real time clock alarm is
-                // a feature that can be used to allow a computer to 'wake up' after shut down to execute
-                // tasks every day or on a certain day. It can sometimes be found in the 'Power Management'
-                // section of a motherboard's BIOS setup. This is functionality that extends beyond
-                // Firecracker intended use. However, we increment a metric just in case.
-                self.match_value = val;
-                METRICS.rtc.missed_write_count.inc();
-            }
-            RTCLR => {
-                self.load = val;
-                self.previous_now = Instant::now();
-                // If the unwrap fails, then the internal value of the clock has been corrupted and
-                // we want to terminate the execution of the process.
-                self.tick_offset = utils::time::seconds_to_nanoseconds(i64::from(val))
-                    .expect("Time conversion overflow");
-            }
-            RTCIMSC => {
-                self.imsc = val & 1;
-            }
-            RTCICR => {
-                // The RTCICR is used together with the RTCMR register to implement a basic time alarm function.
-                // As per the above comment, we do not implement this functionality in firecracker, so self.ris
-                // never gets asserted. Just like we do
-                // for the RTCMR, we set the expected value and we increment a missed metric.
-                self.ris &= !val;
-                METRICS.rtc.missed_write_count.inc();
-            }
-            RTCCR => (), // ignore attempts to turn off the timer.
-            off => {
-                return Err(Error::BadOffset(off, "write"));
-            }
-        }
-        Ok(())
-    }
-}
-
-impl BusDevice for Rtc {
+#[cfg(target_arch = "aarch64")]
+impl BusDevice for RTC<Arc<RTCDeviceMetrics>> {
     fn read(&mut self, offset: u64, data: &mut [u8]) {
         if data.len() == 4 {
-            match self.handle_read(offset) {
-                Ok(val) => byte_order::write_le_u32(data, val),
-                Err(e) => {
-                    warn!("Failed to read from the RTC: {}", e);
-                    METRICS.rtc.error_count.inc();
-                }
-            }
+            // read() expects a slice of len 4, and we just validated
+            // that this is the data length
+            self.read(offset as u16, data.try_into().unwrap())
```

Comment:
I would clarify the comments above: we are calling the RTC implementation. `RTC::read()`